app/components/assign-students.js

```diff
@@ -22,20 +22,22 @@ export default Component.extend({
   selectedUserIds: null,
 
   cohorts: reads('loadCohorts.lastSuccessful.value'),
+  isLoading: reads('loadCohorts.isRunning'),
 
-  bestSelectedCohort: computed('cohorts.[]', 'primaryCohortId', function() {
-    return new RSVP.Promise(resolve => {
-      const primaryCohortId = this.primaryCohortId;
-      const cohorts = this.cohorts;
-      if (primaryCohortId) {
-        let currentCohort = cohorts.find(cohort => cohort.id === primaryCohortId);
-        if (currentCohort) {
-          resolve(currentCohort);
-          return;
-        }
-      }
-      resolve(cohorts.get('lastObject'));
-    });
+  bestSelectedCohort: computed('cohorts', 'primaryCohortId', function() {
```

Comment:
Should be `cohorts.[]` right?

Reply:
The promise never resolves as cohorts are empty initially so I put a check to see if cohorts are there or not. I was thinking just `cohorts` because the newly ran task will constantly replace it, instead of adding/removing to an array.

https://github.com/ilios/frontend/blob/master/app/components/assign-students.js#L29

Reply:
The next person who touches this code may make changes to this though, and the person after that. They may not realized they need to add `.[]` here. Is there some gain in not identifying the array elements to be tracked?

Reply:
I don't have a strong opinion at all on this so I am adding it back. 👍